src/test/java/com/cag/cagbackendapi/integration/UserControllerIntegrationTests.kt

```diff
@@ -0,0 +1,82 @@
+package com.cag.cagbackendapi.integration
+
+import com.cag.cagbackendapi.constants.DetailedErrorMessages
+import com.cag.cagbackendapi.constants.RestErrorMessages
+import com.cag.cagbackendapi.error.ErrorDetails
+import com.cag.cagbackendapi.error.exceptions.UnauthorizedException
+import com.cag.cagbackendapi.model.UserModel
+import com.cag.cagbackendapi.util.SpringCommandLineProfileResolver
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.web.client.TestRestTemplate
+import org.springframework.http.*
+import org.springframework.test.context.ActiveProfiles
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles(value = ["LOCAL"], resolver = SpringCommandLineProfileResolver::class)
+class UserControllerIntegrationTests {
+
+    @Autowired
+    lateinit var testRestTemplate: TestRestTemplate
+
+    private val objectMapper = ObjectMapper()
+
+    @Test
+    fun registerUser_validInput_201Success() {
+        val testUser = UserModel("john testy", "jj@aol.com")
+
+        val headers = HttpHeaders()
+        headers.set("authKey", "mockAuthKey")
+        val request = HttpEntity(testUser, headers)
+
+        val result = testRestTemplate.postForEntity("/user/register", request, String::class.java)
+
+        assertNotNull(result)
+        assertEquals(HttpStatus.CREATED, result.statusCode)
+        assertEquals(testUser, request.body)
+    }
+
+    @Test
+    fun registerUser_emptyMessage_401Unauthorized() {
+        val testUser = UserModel("john testy", "jj@aol.com")
+
+        val headers = HttpHeaders()
+        headers.set("authKey", "wrongAuthKey")
+        val request = HttpEntity(testUser, headers)
+
+        val result = testRestTemplate.postForEntity("/user/register", request, ResponseEntity::class.java)
+
+        //assertEquals(HttpStatus.UNAUTHORIZED, result.)
+        assertNotNull(request.body)
+//        assertEquals(actual.restErrorMessage, RestErrorMessages.UNAUTHORIZED_MESSAGE)
```

Comment:
Can we remove this commented out code until we need it, as well

Reply:
This is work in progress, I was working on this immediately before I ran out of time and just pushed all these changes up. This is actually incomplete code but it's enough to get us started and I'll continue from here once I get back to it. I am against pushing incomplete code 100% but it's gonna take a week for me to finish setting up the database and fixing tests and docker stuff. I want our team to start to understand the structure of our project rather than wait another week fro me to finish the setup. Won't happen often!